### DISCUSSION TITLE
Harfuzz is requesting kerning between ligature glyph and 0xFFFF(AAT::DELETED_GLYPH) for ttc font in mac

### DISCUSSION THREAD

**Discussion by DeependraKhetwal** [harfbuzz/harfbuzz#4597] [Q&A] [answered]
Hello we am using harfbuzz with our custom font fuction for getting glyph kerning ( hb_font_get_glyph_h_kerning_func_t) . On mac while we are applying ligatures (for example ff) with lucida grande font, then harfuzz is calling our custom font function for kerning between ligature (gid: 1277) and gid:0xFFFF(AAT::DELETED_GLYPH) . It seems harfuzz is not removing the deleted glyphs in https://github.com/harfbuzz/harfbuzz/blob/main/src/hb-ot-shape.cc#L956 ( because c->plan->apply_gpos is false) before calling the custom kerning function during hb_ot_position. This is resulting in exception as the font function is not able to process the gid:0xFFFF(AAT::DELETED_GLYPH).

**Comment by behdad** [comment, marked as answer]
First, I'm really surprised that Lucida Grande has no kerning info itself!
Second, your system should be resistant to missing glyphs since fonts can produce those.

I'll see about fixing the bug though.

I'm also curious; what does your custom kerning function do?

Lastly, please remove the uploaded font since it's proprietary.

**Reply by behdad** [reply]
I pushed a fix to main. Please test.

**Reply by khaledhosny** [reply]
> First, I'm really surprised that Lucida Grande has no kerning info itself!

IIRC, All Lucida fonts have no kerning, they were designed to not need kerning.

**Reply by DeependraKhetwal** [reply]
@behdad / @khaledhosny  I checked the issue is fixed. Thanks a lot for the prompt resolution
